tests/apps/namd/namd_test.py

```diff
@@ -0,0 +1,171 @@
+        self.num_tasks = int(self.num_nodes)
+
+        configFile = self.download_material()
+
+        # VSC specific config
+        if self.current_system.name == 'leibniz':
+            self.num_cpus_per_task = 28
+            self.modules = ['NAMD/2.14-verbs-smp']
+        elif self.current_system.name == 'vaughan':
+            self.num_cpus_per_task = 64
+            self.modules = ['NAMD/2.14-verbs-smp']
+        elif self.current_system.name == 'genius':
+            self.num_cpus_per_task = 36
+            self.modules = ['NAMD-prebuilt/2.13-intel-2018a-verbs-smp']
+
+        self.create_nodelist()
+
+        self.executable = f'charmrun ++p {self.num_cpus_per_task*self.num_tasks} ++ppn {self.num_cpus_per_task} ++nodelist mynodes $EBROOTNAMD/namd2 {configFile}/{configFile}.namd'
+
+
+@rfm.simple_test
+class Namd_NotSMP_CPUTest(NamdBaseTest):
+    # NAMD notSMP CPU test
+
+    def __init__(self):
+        self.time_limit = '20m'
+
+        self.valid_systems = ['leibniz:single-node',
+                              'vaughan:single-node',
+                              'genius:single-node',
+                              'hydra:single-node']
+
+        self.valid_prog_environs = ['builtin']
+        super().__init__('cpu')
+
+        self.scale_reference = {
+            '1': {
+                'leibniz:single-node': {'days_ns': (0.347779, None, 0.05, 'days/ns')},
+                'vaughan:single-node': {'days_ns': (0.188093, None, 0.05, 'days/ns')},
+                'hydra:single-node': {'days_ns': (0.207591, None, 0.05, 'days/ns')},
+            },
+            '2': {
+                'leibniz:single-node': {'days_ns': (0.1782715, None, 0.05, 'days/ns')},
+                'vaughan:single-node': {'days_ns': (0.09856985, None, 0.05, 'days/ns')},
+                'hydra:single-node': {'days_ns': (0.1704705, None, 0.05, 'days/ns')},
+            },
+            '4': {
+                'leibniz:single-node': {'days_ns': (1.05726, None, 0.05, 'days/ns')},
+                'vaughan:single-node': {'days_ns': (0.5438339, None, 0.05, 'days/ns')},
+                'hydra:single-node': {'days_ns': (4.24083, None, 0.05, 'days/ns')},
+            },
+        }
+        self.reference = self.scale_reference[self.num_nodes]
+
+    @run_after('setup')
+    def set_num_cpus(self):
+        self.num_tasks = int(self.num_nodes)
+
+        configFile = self.download_material()
+
+        launcher = 'charm'
+
+        # VSC specific config
+        if self.current_system.name == 'leibniz':
+            self.num_cpus_per_task = 28
+            self.modules = ['NAMD/2.14-verbs']
+        elif self.current_system.name == 'vaughan':
+            self.num_cpus_per_task = 64
+            self.modules = ['NAMD/2.14-verbs']
+        elif self.current_system.name == 'hydra':
+            self.num_tasks = 40 * self.num_tasks
+            self.modules = ['NAMD/2.14-foss-2019b-mpi']
+            self.job.options = ["--partition=skylake,skylake_mpi",
```

Comment:
for multi-node jobs you need to use skylake_mpi to get infiniband interconnect
```suggestion
            self.job.options = ["--partition=skylake_mpi",
```

Reply:
LGTM